hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/ConsistentHashingBucketInsertPartitioner.java

```diff
@@ -29,5 +29,14 @@ public interface ConsistentHashingBucketInsertPartitioner {
    * @param partition partition to set Consistent Hashing nodes
    * @param nodes     nodes from clustering plan
    */
+  /**
+   * Set pending consistent hashing for partition, only used in executing clustering
+   * When call this method, the bulk insert will directly use the pending metadata as the consistent hash metadata for writing data to after-resizing buckets.
+   * Used in the case of executing bulk insert.
+   * NOTE: This method should be called before the bulk insert operation, and will skip building identifiers from records, just use the pending metadata.
+   * For which not calling this method, the bulk insert will use the committed metadata as the bucket metadata and disallow writing data to the pending-resizing buckets.
+   * @param partition partition to set Consistent Hashing nodes
+   * @param nodes     nodes from clustering plan
+   */
```

Comment:
Does one partitioner instance works for only one partition during the whole life-cycle?

Reply:
Then shall we move the instantiation of the metadata once in the constructor? Not sure if there are some SE/DE issue.

Reply:
> Then shall we move the instantiation of the metadata once in the constructor? Not sure if there are some SE/DE issue.

What you means is that we remove the method: `ConsistentHashBucketInsertPartitioner::addHashingChildrenNodes`, and in the clustering scenario, the partitioner is instantiated directly by carrying these metadata as construct parameters?

Reply:
yeah, if the SE/DE works well, maybe we just put the metadata in the partitioner constructor.

Reply:
> yeah, if the SE/DE works well, maybe we just put the metadata in the partitioner constructor.

Done~ I've refactored this part of the code.